GoogleForCreators/web-stories-wp
# Compatibility with hotlinked story posters

## Context



## Summary

Hook into `has_post_thumbnail` and `post_thumbnail_html` filters to support, hotlinked poster images. 

## Relevant Technical Choices

Force the height and width to be the request height and width. As the hotlinked image will not have different sizes, we can only use the full size. 

## To-do



## User-facing changes

<img width="2119" alt="Screenshot 2022-10-13 at 15 55 30" src="https://user-images.githubusercontent.com/237508/195632175-cfd44759-fd13-40f0-a773-71360711ee64.png">

## Testing Instructions






This PR can be tested by following these steps:

1. Create a story. 
2. Add a hotlinked story image. Example. https://assets.computingcorner.co.uk/uploads/2022/10/example-3.png
3. Publish story. 
4. Go to FE of site. <site_url>/web-stories
5. See poster image used on as featured image. 

Try this on core themes, by changing site theme to 2010-2023 themes. 


## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #12028
